### PR TITLE
Original weakness

### PR DESCRIPTION
**Pull request by Rvtar**
An implementation of weakness that matches how it works on Hermitcraft. The weakness icons will show up at hermits with types that have their weaknesses modified.

The feedback was kinda mixed so I might as well try.

### CODE REVIEW

**Comment by Rvtar** [comment]
Just realized that onCardPlay is just a bootleg onAttach that gives the whole card instead of just cardInstance. Gonna leave it on for now because it makes this thing a whole lot simpler and changing onAttach to input that would probably be a refactor a that point.

**Comment by Sense101** [comment]
Yeah we don't want to be adding unnecessary hooks, I'm sure you can find a way to do this with onAttach (e.g. is there a way you could use the card instance to get more information about the card?)

**Comment by Benji42** [comment]
`getCardPos` is probably what you want.

**Comment by Rvtar** [comment]
Yeah, I completely missed it the first time.

const card = gatCardAtPos(game, getBasicCardPos(game, instance)) should work.

**Comment by Rvtar** [comment]
(done)

**Comment by Rvtar** [comment]
1) I have prettier installed.
2) I formatted all of the listed documents.

What is going on?

**Comment by Rvtar** [comment]
It's good now.

**Comment by Sense101** [comment]
Why are you calling it weakness dummy? Can you please clarify what exactly is changed by moving over weakness to the original?

**Comment by Rvtar** [comment]
There's a brain part that is invisible and the UI (dummy) that's visible. Cannot have both properties at the same time. I'm not sure of applying the brain to oblivion would cause any issues but if doable, it can be reduced to a single one.

**Comment by Rvtar** [comment]
For the mechanics, the strengths and weaknesses are tied to the types themselves rather than the hermits.

**Comment by Benji42** [comment]
This feels a bit overly-complicated to me. Why can't the weakness dummy stuff be on the weakness potion?

But to be honest, I think the real change that needs to be made here is to allow status effects to have multiple targets. It would allow this to be implemented a lot nicer, I think. I would be happy with it then.

File: common/status-effects/weaknessRef.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+					}
+				}
+			}
+		}
+
+		if (!statusEffectInfo.duration) statusEffectInfo.duration = this.duration
+
+		applyDummies()
+
+		STRENGTHS[strongType].push(weakType)
```
Comment (Lunarmagpie) [inline common/status-effects/weaknessRef.ts:68]: This doesn't work. This modifies the weak types for everyone changing the game.
This variable is supposed to be immutable.

**Comment by Lunarmagpie** [comment]
I suggest implementing this by changing the weakness of the incoming attacks  to the hermit to `always`.

**Comment by Rvtar** [comment]
I realized I shouldn't have pushed this stuff rn.

**Comment by Rvtar** [comment]
I'll open again when I'm done.

**Pull request closed without merging**